k8s/controllers/solution/instance_controller.go

```diff
@@ -138,7 +138,11 @@ func (r *InstanceReconciler) deploymentBuilder(ctx context.Context, object recon
 		TargetCandidates: []fabric_v1.Target{},
 	}
 
-	if err := r.Get(ctx, types.NamespacedName{Name: instance.Spec.Solution, Namespace: instance.Namespace}, &deploymentResources.Solution); err != nil {
+	solutionName := instance.Spec.Solution
+	if strings.Contains(solutionName, ":") {
+		solutionName = strings.ReplaceAll(solutionName, ":", "-")
```

Comment:
I wish you can write an util function to hide the string replacement in business logic, which is easily to reuse and maintain in future.

Reply:
Can do for `replaceColon` function in k8s util.